File: Java/src/main/java/QuestionBank/Solution901_1000/Solution905.java
package QuestionBank.Solution901_1000;

/**
 * @author windows
 */
public class Solution905 {
    public int[] sortArrayByParity(int[] nums) {
        int left = 0;
        int right = nums.length - 1;
        while (left < right){
            while (left < right && nums[left] % 2 != 0){
                 int temp = nums[left];
                 nums[left] = nums[right];
                 nums[right] = temp;
                 right--;
            }
            left++;
        }
        return nums;
    }


    public static void main(String[] args) {
        Solution905 solution905 = new Solution905();
    }
}

File: C++/src/Solution901_1000/Solution905.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution905.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> Solution::sortArrayByParity(vector<int> &nums) {
    int n = nums.size();
    int left = 0;
    for (int i = 0; i < n; ++i) {
        if (nums[i] % 2 == 0){
            int temp = nums[i];
            nums[i] = nums[left];
            nums[left] = temp;
            left++;
        }
    }
    return nums;
}
